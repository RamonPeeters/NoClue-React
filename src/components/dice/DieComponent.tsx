import React, { Component, ReactNode } from "react";
import { Image, ImageURISource, StyleProp, StyleSheet, ViewStyle } from "react-native";

interface Props {
    value: number;
    style?: StyleProp<ViewStyle>;
}

export default class DieComponent extends Component<Props> {
    private static readonly ONE: ImageURISource = require('../../assets/die/1.png');
    private static readonly TWO: ImageURISource = require('../../assets/die/2.png');
    private static readonly THREE: ImageURISource = require('../../assets/die/3.png');
    private static readonly FOUR: ImageURISource = require('../../assets/die/4.png');
    private static readonly FIVE: ImageURISource = require('../../assets/die/5.png');
    private static readonly SIX: ImageURISource = require('../../assets/die/6.png');

    public render(): ReactNode {
        return (
            <Image style={[STYLES.container, this.props.style]} source={DieComponent.getDieImage(this.props.value)}></Image>
        );
    }

    private static getDieImage(value: number): ImageURISource {
        switch (value) {
            case 1: {
                return DieComponent.ONE;
            }
            case 2: {
                return DieComponent.TWO;
            }
            case 3: {
                return DieComponent.THREE;
            }
            case 4: {
                return DieComponent.FOUR;
            }
            case 5: {
                return DieComponent.FIVE;
            }
            case 6:
            default: {
                return DieComponent.SIX;
            }
        }
    }
}

const STYLES = StyleSheet.create({
    container: {
    }
});
