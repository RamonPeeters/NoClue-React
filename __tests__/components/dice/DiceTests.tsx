import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import DieComponent from '../../../src/components/dice/DieComponent';

it('renders correct image for die value 1', () => {
    const dieComponent = renderer.create(<DieComponent value={1} />);
    expect(dieComponent).toMatchSnapshot();
});

it('renders correct image for die value 2', () => {
    const dieComponent = renderer.create(<DieComponent value={1} />);
    expect(dieComponent).toMatchSnapshot();
});

it('renders correct image for die value 3', () => {
    const dieComponent = renderer.create(<DieComponent value={2} />);
    expect(dieComponent).toMatchSnapshot();
});

it('renders correct image for die value 4', () => {
    const dieComponent = renderer.create(<DieComponent value={4} />);
    expect(dieComponent).toMatchSnapshot();
});

it('renders correct image for die value 5', () => {
    const dieComponent = renderer.create(<DieComponent value={5} />);
    expect(dieComponent).toMatchSnapshot();
});

it('renders correct image for die value 6', () => {
    const dieComponent = renderer.create(<DieComponent value={6} />);
    expect(dieComponent).toMatchSnapshot();
});

it('renders the default image for an invalid die value', () => {
    const dieComponent = renderer.create(<DieComponent value={-1} />);
    expect(dieComponent).toMatchSnapshot();
});
