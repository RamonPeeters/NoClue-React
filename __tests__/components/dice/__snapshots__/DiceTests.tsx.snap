// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct image for die value 1 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/1.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders correct image for die value 2 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/1.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders correct image for die value 3 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/2.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders correct image for die value 4 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/4.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders correct image for die value 5 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/5.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders correct image for die value 6 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/6.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;

exports[`renders the default image for an invalid die value 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/die/6.png",
    }
  }
  style={
    Array [
      Object {},
      undefined,
    ]
  }
/>
`;
